app/assets/javascripts/discourse/app/components/time-shortcut-picker.js

```diff
@@ -197,34 +198,40 @@ export default Component.extend({
       });
     }
 
-    if (hiddenOptions.length > 0) {
-      options.forEach((opt) => {
-        if (hiddenOptions.includes(opt.id)) {
-          opt.hidden = true;
-        }
-      });
-    }
+    customOptions.forEach((opt) => {
+      if (!opt.timeFormatted && opt.time) {
+        opt.timeFormatted = opt.time.format(I18n.t(opt.timeFormatKey));
+      }
+    });
+
+    options = options.concat(customOptions);
+    options.sort((a, b) => {
+      return a.time < b.time ? -1 : a.time > b.time ? 1 : 0;
```

Comment:
Can we avoid nesting two inlines if here? It is quite hard to follow.

Reply:
@tgxworld fixed now

Reply:
I find the JS sorting thing annoying to read as well, but yeah the double ternery was too much